itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java

```diff
@@ -589,6 +590,72 @@ public void testIncrementalAdds() throws IOException {
   }
 
   @Test
+  public void testIncrementalLoadWithVariableLengthEventId() throws IOException, TException {
```

Comment:
I think. this test will have other issues as well. If we set to 1 (anything less than lastReplId of bootstrap), then, it is possible that old events gets applied twice due to filter logic and cause inconsistency. Need to see, if this test case can be removed as it doesn't make sense in future. However, I add unit test to verify the sort order of event IDs with new comparator.